src/parser/jobs/rpr/modules/HarvestMoon.tsx

```diff
@@ -0,0 +1,90 @@
+import {Trans} from '@lingui/react'
+import {DataLink} from 'components/ui/DbLink'
+import {Event} from 'event'
+import {Analyser} from 'parser/core/Analyser'
+import {filter} from 'parser/core/filter'
+import {dependency} from 'parser/core/Injectable'
+import Checklist, {Requirement, Rule} from 'parser/core/modules/Checklist'
+import {Data} from 'parser/core/modules/Data'
+import {Invulnerability} from 'parser/core/modules/Invulnerability'
+import {UnableToAct} from 'parser/core/modules/UnableToAct'
+import React from 'react'
+
+const SOULSOW_BUFFER = 1000
+
+export class HarvestMoon extends Analyser {
+	static override handle = 'harvestMoon'
+
+	@dependency private checklist!: Checklist
+	@dependency private data!: Data
+	@dependency private invulnerability!: Invulnerability
+	@dependency private unableToAct!: UnableToAct
+
+	private harvestMoonCasts = 0
+
+	override initialise() {
+		super.initialise()
+
+		this.addEventHook(
+			filter<Event>()
+				.source(this.parser.actor.id)
+				.type('action')
+				.action(this.data.actions.HARVEST_MOON.id),
+			() => this.harvestMoonCasts++)
+
+		this.addEventHook('complete', this.onComplete)
+	}
+
+	private windowFilterCallback(inputWindow: {start: number, end: number}): boolean {
+		const unableToActWindow = this.unableToAct.getWindows(inputWindow)[0]
+
+		if (unableToActWindow == null) {
+			return false
+		}
+
+		if (unableToActWindow.start - inputWindow.start >= this.data.actions.SOULSOW.castTime + SOULSOW_BUFFER) {
+			return true
+		}
```

Comment:
This is used a few times, may just wanna const the cast time with buffer. You can also invert this so it can be a single conditional with the previous one.
```suggestion
		if ((unableToActWindow == null || unableToActWindow.start - inputWindow.start < this.data.actions.SOULSOW.castTime + SOULSOW_BUFFER) {
			return false
		}
```